pennylane/tape/tapes/tape.py

```diff
@@ -814,6 +818,71 @@ def graph(self):
 
         return self._graph
 
+    @property
+    def resources(self):
+        """Resource requirements of a quantum circuit.
+
+        Returns:
+            dict[str, int]: how many times constituent operations are applied
+
+        **Example**
+
+        .. code-block:: python3
+
+            with qml.tape.QuantumTape() as tape:
+                qml.Hadamard(wires=0)
+                qml.RZ(0.26, wires=1)
+                qml.CNOT(wires=[1, 0])
+                qml.Rot(1.8, -2.7, 0.2, wires=0)
+                qml.Hadamard(wires=1)
+                qml.CNOT(wires=[0, 1])
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        Asking for the resources produces a dictionary as shown below:
+
+        >>> tape.resources
+        {'Hadamard': 2, 'RZ': 1, 'CNOT': 2, 'Rot': 1}
+        """
+        if self._resources is None:
+            self._resources = {}
+
+            for op in self.operations:
+                if op.name not in self._resources.keys():
+                    self._resources[op.name] = 1
+                else:
+                    self._resources[op.name] += 1
+
+        return self._resources
+
+    @property
+    def depth(self):
+        """Depth of the quantum circuit.
+
+        Returns:
+            int: Circuit depth, computed as the longest path in the circuit's directed acyclic graph representation.
+
+        **Example**
+
+        .. code-block:: python3
+
+            with QuantumTape() as tape:
+                qml.Hadamard(wires=0)
+                qml.PauliX(wires=1)
+                qml.CRX(2.3, wires=[0, 1])
+                qml.Rot(1.2, 3.2, 0.7, wires=[1])
+                qml.CRX(-2.3, wires=[0, 1])
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        Asking for the resources produces a dictionary as shown below:
```

Comment:
copy/paste mismatch? "shown below" seems to not point to anything :thinking:, and this is the `depth` function, not the `resources` function